packages/typespec-test/test/openai_modular/generated/typespec-ts/src/models/models.ts

```diff
@@ -532,101 +1316,334 @@ export interface ChatCompletions {
+/**
+ * A representation of a single generated image, provided as either base64-encoded data or as a URL from which the image
+ * may be retrieved.
+ */
+export interface ImageGenerationData {
+  /** The URL that provides temporary access to download the generated image. */
+  url?: string;
+  /** The complete data for an image, represented as a base64-encoded string. */
+  base64Data?: string;
+  /**
+   * The final prompt used by the model to generate the image.
+   * Only provided with dall-3-models and only when revisions were made to the prompt.
+   */
+  revisedPrompt?: string;
+}
+
+/**
+ * The configuration information for an embeddings request.
+ * Embeddings measure the relatedness of text strings and are commonly used for search, clustering,
+ * recommendations, and other similar scenarios.
+ */
+export interface EmbeddingsOptions {
+  /**
+   * An identifier for the caller or end user of the operation. This may be used for tracking
+   * or rate-limiting purposes.
+   */
+  user?: string;
+  /**
+   * The model name to provide as part of this embeddings request.
+   * Not applicable to Azure OpenAI, where deployment information should be included in the Azure
+   * resource URI that's connected to.
+   */
+  model?: string;
+  /**
+   * Input texts to get embeddings for, encoded as a an array of strings.
+   * Each input must not exceed 2048 tokens in length.
+   *
+   * Unless you are embedding code, we suggest replacing newlines (\n) in your input with a single space,
+   * as we have observed inferior results when newlines are present.
+   */
+  input: string[];
+}
+
+/**
+ * Representation of the response data from an embeddings request.
+ * Embeddings measure the relatedness of text strings and are commonly used for search, clustering,
+ * recommendations, and other similar scenarios.
+ */
+export interface Embeddings {
+  /** Embedding values for the prompts submitted in the request. */
+  data: EmbeddingItem[];
+  /** Usage counts for tokens input using the embeddings API. */
+  usage: EmbeddingsUsage;
+}
+
+/** Representation of a single embeddings relatedness comparison. */
+export interface EmbeddingItem {
+  /**
+   * List of embeddings value for the input prompt. These represent a measurement of the
+   * vector-based relatedness of the provided input.
+   */
+  embedding: number[];
+  /** Index of the prompt to which the EmbeddingItem corresponds. */
+  index: number;
+}
+
+/** Measurement of the amount of tokens used in this request and response. */
+export interface EmbeddingsUsage {
+  /** Number of tokens sent in the original request. */
+  promptTokens: number;
+  /** Total number of tokens transacted in this request/response. */
+  totalTokens: number;
+}
+
+/** Alias for ChatRequestMessage */
+export type ChatRequestMessage =
+  | ChatRequestSystemMessage
+  | ChatRequestUserMessage
+  | ChatRequestAssistantMessage
+  | ChatRequestToolMessage
+  | ChatRequestFunctionMessage
+  | ChatRequestMessageParent;
```

Comment:
Put a pin here, we have to use Union as the suffix so that we are consistent with hlc.